cmdx.py

```diff
@@ -4096,9 +4192,38 @@ def createNode(self, type, name=None, parent=None):
         return DagNode(mobj, exists=False, modifier=self)
 
     @record_history
-    def parent(self, node, parent=None):
-        parent = parent._mobject if parent is not None else None
-        self._modifier.reparentNode(node._mobject, parent)
+    def parent(self, node, parent=None, preserve=True):
+
+        if preserve:
+            if parent is None:
+                transform = node.transform(space=sWorld)
+            else:
+                transform = node.mapFrom(parent)
+
+        _parent = parent._mobject if parent is not None else om.MObject.kNullObj
+        self._modifier.reparentNode(node._mobject, _parent)
+
+        if preserve:
+            self.transform(node, transform)
+
+        if node.typeId == tJoint:
+            self.disconnect(node['inverseScale'])
+            if parent and parent.typeId == tJoint:
+                self.connect(parent['scale'], node['inverseScale'])
+
+    @record_history
+    def transform(self, node, transform):
+
+        if transform.rotationOrder() - 1 != node['ro']:
```

Comment:
Need to document what's going on here. Why -1?

Also try and stick with double-quotes `"` instead of `'`, for consistency.

Reply:
ah yes sorry i forgot that one. it is simply a mapping between the rotateOrder plug from transform nodes and the rotationOrder from transformationMatrix

not very convenient.. (: